src/cargo/core/dependency.rs

```diff
@@ -240,8 +282,12 @@ impl Dependency {
         self.inner.platform.as_ref()
     }
 
-    pub fn rename(&self) -> Option<&str> {
-        self.inner.rename.as_ref().map(|s| &**s)
+    /// The renamed name of this dependency, if any.
+    ///
+    /// If the `package` key is used in `Cargo.toml` then this returns the same
+    /// value as `name_in_toml`.
```

Comment:
I don't want to stall this PR, but after several attempts I still can't wrap my head around this explanation.

Reply:
Ah sorry about that! I'll try to get around to cleaning this up soon.